### PR TITLE
Adds trace id logging and propagation

### PR DESCRIPTION
**Pull request by christopherjturner**
Also some general cleanup/lint fixing

### CODE REVIEW

File: Backend.Api/Config/Environment.cs [hunk 1]
```diff
@@ -6,4 +6,4 @@ public static bool IsDevMode(this WebApplicationBuilder builder)
     {
         return !builder.Environment.IsProduction();
     }
-}
+}
```
Comment (flurdy) [inline Backend.Api/Config/Environment.cs:9]: Not the biggest change...

File: Backend.Api/Program.cs [hunk 2]
```diff
@@ -2,94 +2,81 @@
...
-
-//-------- Configure the WebApplication builder------------------//
+using Backend.Api.Config;
+using Backend.Api.Utils.Logging;
+using Serilog;
 
 var app = CreateWebApplication(args);
 await app.RunAsync();
-
+return;
```
Comment (flurdy) [inline Backend.Api/Program.cs:14]: Does this do what is intended?

**Review by flurdy — APPROVED** [review]
Looks alright. 
Your formatter seem to remove the last empty line in files though?

File: Backend.Api/Config/Environment.cs (hunk 1 above)
Comment (christopherjturner) [reply Backend.Api/Config/Environment.cs:9]: Re-ran auto-formatter. I think one or two of the files were in a non-utf file format and/or had 3 space indentation...

Comment (christopherjturner) [reply Backend.Api/Config/Environment.cs:9]: wonder if we need to figure out which auto-formatter to use?

File: Backend.Api/Program.cs (hunk 2 above)
Comment (christopherjturner) [reply Backend.Api/Program.cs:14]: The linter demands it!
Basically the new style program.cs entrypoint/main would rather you just had no additonal methods etc defined so its just linear. 
The return just makes it explicit where the entrypoint ends.

**Comment by christopherjturner** [comment]
I think we might need to look at linting/formatting, feels like we have a few conflicting tools here. (e.g. var _privateField vs var privateField)

**Pull request merged**